### PR TITLE
driver spectrometer: do software binning if hardware doesn't support full vertical binning

### PR DESCRIPTION
**Pull request by pieleric**

### CODE REVIEW

File: src/odemis/driver/spectrometer.py [hunk 1]
```diff
@@ -273,8 +275,25 @@ def _newFrame(self, df, data):
         Get the new frame from the detector
         """
         if data.shape[0] != 1:
```
Comment (Sabrina85) [inline src/odemis/driver/spectrometer.py]: data.shape[0] is horizontal? Should be vertical? Or is that inverted for data?

File: src/odemis/driver/test/spectrometer_test.py [hunk 2]
```diff
@@ -252,6 +255,27 @@ def test_vbinning(self):
         md = data.metadata
         self.assertEqual(md[model.MD_BINNING], tuple(binning))
 
+        # reduce the binning to just the CCD max binning
```
Comment (Sabrina85) [inline src/odemis/driver/test/spectrometer_test.py]: It would be nice to have a bit more info what it tested for, as it is hard to follow that way. The binning of the spectrometer?

**Comment by pieleric** [comment]
Updated to make the comments clearer.
